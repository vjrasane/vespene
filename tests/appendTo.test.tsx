import V, { render } from "../src";

describe("appendTo", () => {
  it("appends unmounted element to mounted element", () => {
    const parent = <div id="parent"></div>;
    render(document.body, parent);
    (<div id="child"></div>).appendTo(parent);
    expect(document.body).toMatchSnapshot();
  });

  it("appends mounted element to mounted element", () => {
    const child1 = <div id="child1"></div>;
    const child2 = <div id="child2"></div>;
    render(
      document.body,
      <div id="parent">
        {child1}
        {child2}
      </div>
    );
    child2.appendTo(child1);
    expect(document.body).toMatchSnapshot();
  });

  it("does not append mounted element to unmounted element", () => {
    const elem1 = <div id="elem1"></div>;
    const elem2 = <div id="elem2"></div>;
    render(document.body, elem1);
    elem1.appendTo(elem2);
    expect(document.body.childElementCount).toBe(1);
    render(document.body, elem2);
    expect(document.body).toMatchSnapshot();
  });

  it("does not append unmounted element to unmounted element", () => {
    const elem1 = <div id="elem1"></div>;
    const elem2 = <div id="elem2"></div>;
    elem1.appendTo(elem2);
    render(document.body, elem2);
    expect(document.body).toMatchSnapshot();
  });

  it("throws an error when appending parent to its child", () => {
    const child = <div id="child"></div>;
    const parent = <div id="parent">{child}</div>;
    render(document.body, parent);
    expect(() => parent.appendTo(child)).toThrowError(/incorrect node tree/);
    expect(document.body).toMatchSnapshot();
  });

  it("throws an error when appending element to itself", () => {
    const elem = <div id="elem"></div>;
    render(document.body, elem);
    expect(() => elem.appendTo(elem)).toThrowError(/incorrect node tree/);
    expect(document.body).toMatchSnapshot();
  });
});
