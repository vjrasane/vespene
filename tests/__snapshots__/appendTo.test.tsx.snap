// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appendTo appends mounted element to mounted element 1`] = `
<body>
  <div
    id="parent"
  >
    <div
      id="child1"
    >
      <div
        id="child2"
      />
    </div>
  </div>
</body>
`;

exports[`appendTo appends mounted element to unmounted element 1`] = `
<body>
  <div
    id="elem2"
  >
    <div
      id="elem1"
    />
  </div>
</body>
`;

exports[`appendTo appends unmounted element to mounted element 1`] = `
<body>
  <div
    id="parent"
  >
    <div
      id="child"
    />
  </div>
</body>
`;

exports[`appendTo appends unmounted element to unmounted element 1`] = `
<body>
  <div
    id="elem2"
  >
    <div
      id="elem1"
    />
  </div>
</body>
`;

exports[`appendTo throws an error when appending element to itself 1`] = `
<body>
  <div
    id="elem"
  />
</body>
`;

exports[`appendTo throws an error when appending parent to its child 1`] = `
<body>
  <div
    id="parent"
  >
    <div
      id="child"
    />
  </div>
</body>
`;
